channel-sdk/src/main/java/com/zello/channel/sdk/image/ImageMessageManager.kt

```diff
@@ -100,13 +120,54 @@ internal class ImageMessageManagerImpl(private val listener: ImageMessageManager
 						imageInfo.thumbnail = decoded
 						incomingImages[imageId] = imageInfo
 						message = ImageInfo(imageId, imageInfo.sender, decoded, imageInfo.image)
+						setNeedsCleanup()
 					}
 				}
 				listener.onImageMessage(message)
 			}
 		}
 	}
 
+	/// last time we performed a cleanup (ms since system boot)
+	private var lastCleanup = 0L
+	private var needsCleanup = false
+
+	/// Only call `setNeedsCleanup` on main thread
```

Comment:
I believe there is a `@MainThread` annotation for this that'll give lint warnings. Unfortunately in my experience to get really good results you have to go all in annotating everything, because the linter doesn't have that much thread awareness, but thought I'd toss it out as an option. Coroutines might also have some similar annotation

Reply:
Cool, I didn't know that!